src/include/btree_inline.h

```diff
@@ -1424,13 +1425,11 @@ __wt_page_evict_retry(WT_SESSION_IMPL *session, WT_PAGE *page)
 
     /*
      * It is possible that we have not started using the timestamps just yet. So, check for the last
-     * time we evicted only if there is a timestamp set.
+     * time we evicted.
      */
-    if (mod->last_eviction_timestamp != WT_TS_NONE) {
-        __wt_txn_pinned_timestamp(session, &pinned_ts);
-        if (pinned_ts > mod->last_eviction_timestamp)
-            return (true);
-    }
+    __wt_txn_pinned_timestamp(session, &pinned_ts);
```

Comment:
I think this is correct but I am not 100% sure. We had a performance bug in this code before. There is a potential to undo that change here. See WT-6484: https://github.com/wiredtiger/wiredtiger/commit/86659fc959c25be0ae5647fe6d0f0b8a476b6c90

Maybe we should leave it as it is, at worst we save a few cycles of computation, at best we prevent WT-6484 from happening again.

Reply:
We can play safe here and undo the change